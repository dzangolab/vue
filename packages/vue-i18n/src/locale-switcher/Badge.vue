<template>
  <span class="badge"
    >{{ l }} <span v-if="country" class="country">{{ country }}</span></span
  >
</template>

<script lang="ts">
export default {
  name: "LocaleBadge",
};
</script>

<script setup lang="ts">
import { computed } from "vue";

import type { PropType } from "vue";

const props = defineProps({
  locale: {
    required: true,
    type: String as PropType<string>,
  },
});

const l = computed(() => {
  return props.locale.substring(0, 2);
});

const country = computed(() => {
  const tokens = props.locale.split("-");

  if (tokens.length > 1) {
    return tokens[1];
  }

  return undefined;
});
</script>

<style lang="css" scoped>
.country {
  bottom: -0.35rem;
  background: var(--locale-switcher-badge-country-bf-color, #6495ed);
  color: var(--locale-switcher-badge-country-color, #fff);
  font-size: 0.5rem;
  font-weight: bold;
  line-height: 1.25;
  padding: 1px 2px;
  position: absolute;
  right: -0.25rem;
}

.badge {
  align-items: center;
  background-color: var(--locale-switcher-badge-bg-color, #c6deff);
  border-radius: 50%;
  display: flex;
  font-size: 0.9rem;
  height: 2em;
  justify-content: center;
  position: relative;
  text-align: center;
  width: 2em;
}
</style>
