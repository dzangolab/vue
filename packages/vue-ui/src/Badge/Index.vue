<template>
  <span :class="badgeClassName">
    <span v-if="iconLeft || !!slots.iconLeft" class="icon-left">
      <slot name="iconLeft">
        <i :class="iconLeft" />
      </slot>
    </span>

    <div v-if="label" class="label">
      {{ label }}
    </div>

    <span v-if="iconRight || !!slots.iconRight" class="icon-right">
      <slot name="iconRight">
        <i :class="iconRight" />
      </slot>
    </span>
  </span>
</template>

<script lang="ts">
export default {
  name: "BadgeComponent",
};
</script>

<script setup lang="ts">
import { computed, useSlots } from "vue";

const props = defineProps({
  className: {
    default: "",
    type: String,
  },
  color: {
    default: "default",
    type: String,
  },
  iconLeft: {
    default: null,
    type: String,
  },
  iconRight: {
    default: null,
    type: String,
  },
  label: {
    default: null,
    type: [String, Boolean],
  },
  rounded: Boolean,
  severity: {
    default: "primary",
    type: String,
    validator: (value: string) =>
      [
        "primary",
        "secondary",
        "alternate",
        "success",
        "danger",
        "warning",
      ].includes(value),
  },
});

const slots = useSlots();

const badgeClassName = computed(() => {
  return ["badge", props.className, props.rounded && "rounded", props.severity]
    .filter(Boolean)
    .join(" ");
});
</script>
