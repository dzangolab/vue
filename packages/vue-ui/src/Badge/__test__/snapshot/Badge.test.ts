import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import Badge from "../../Index.vue";

describe("Badge", () => {
  it("matches snapshot", () => {
    const wrapper = mount(Badge, {
      props: {
        iconLeft: "pi pi-home",
        iconRight: "pi pi-home",
        label: "Basic",
        class: "rounded",
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
