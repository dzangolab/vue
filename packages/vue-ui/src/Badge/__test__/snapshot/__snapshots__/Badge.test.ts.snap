// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge > matches snapshot 1`] = `
<span
  class="badge rounded primary"
>
  <span
    class="icon-left"
  >
    
    <i
      class="pi pi-home"
    />
    
  </span>
  <div
    class="label"
  >
    Basic
  </div>
  <span
    class="icon-right"
  >
    
    <i
      class="pi pi-home"
    />
    
  </span>
</span>
`;
